### ISSUE TITLE
Unable to use qualified type names

### ISSUE THREAD

**Issue by bgamari** [tdammers/yeshql#9]
Currently it's not possible to refer to a type by its qualified name. For instance,
```haskell
[yesh1|
  -- name:hello :: rowcount Int
  -- :arg    :: Data.Int.Int
  ...
|]
```

**Comment by tdammers** [comment]
Correct; the parser was kept as dumb as humanly possible in order to avoid (unsuccessfully, even, as evidenced by #7) clashes with SQL syntax, and also to avoid having half a Haskell parser embedded in the extended-SQL parser. I think allowing dots wouldn't be a problem though.
